pkgs/development/tools/misc/coccinelle/default.nix

```diff
@@ -1,21 +1,41 @@
-{ fetchurl, lib, stdenv, python3, ncurses, ocamlPackages, pkg-config }:
+{ fetchFromGitHub, fetchpatch, lib, stdenv, python3, ncurses, ocamlPackages, pkg-config, autoreconfHook }:
 
 stdenv.mkDerivation rec {
   pname = "coccinelle";
-  version = "1.1.0";
+  version = "1.1.1";
 
-  src = fetchurl {
-    url = "https://coccinelle.gitlabpages.inria.fr/website/distrib/${pname}-${version}.tar.gz";
-    sha256 = "0k0x4qnxzj8fymkp6y9irggcah070hj7hxq8l6ddj8ccpmjbhnsb";
+  # The release tarball is unable to fetch so use github mirror insead (see NixOS/nixpkgs #171019)
+  src = fetchFromGitHub {
+    repo = pname;
+    rev = version;
+    owner = "coccinelle";
+    hash = "sha256-rS9Ktp/YcXF0xUtT4XZtH5F9huvde0vRztY7vGtyuqY=";
   };
 
-  buildInputs = with ocamlPackages; [
-    ocaml findlib menhir
-    ocaml_pcre parmap stdcompat
-    python3 ncurses pkg-config
+  patches = [
+    # Fix data path lookup.
+    # https://github.com/coccinelle/coccinelle/pull/270
+    (fetchpatch {
+      url = "https://github.com/coccinelle/coccinelle/commit/540888ff426e0b1f7907b63ce26e712d1fc172cc.patch";
+      sha256 = "sha256-W8RNIWDAC3lQ5bG+gD50r7x919JIcZRpt3QSOSMWpW4=";
+    })
+  ];
+
+  nativeBuildInputs = [
+    autoreconfHook
   ];
 
-  doCheck = false;
+  buildInputs = with ocamlPackages; [
+    ocaml
+    findlib
+    menhir
+    ocaml_pcre
+    parmap
+    stdcompat
+    python3
+    ncurses
+    pkg-config
```

Comment:
Many of these belong to `nativeBuildInputs`, try `strictDeps = true;`

Reply:
it turns out, all of them are `nativeBuildInputs`

Reply:
That doesn't sound right, a lot of things in there are linked against for runtime, not merely just executed at build time.

Reply:
I think OCaml tooling might just handling `strictDeps` incorrectly. [When I updated it](https://github.com/NixOS/nixpkgs/commit/276e59eefa0e2677cde603c1312458029ce7b800), I moved few more things to `nativeBuildInputs` – `pkg-config` should definitely be there, and I believe `menhir` is only used to generate parser code at build time. But I am not 100% on the OCaml deps, even though I hunted through the build files quite a bit.

Reply:
I ran `ldd` on executables and `.so` s to find required libraries. seems like `pcre` is required.
ocaml and python should obviously be required I don't know why I forgot.

```
[nix-shell:~/.cache/nixpkgs-review/pr-171019-1]$ ldd ./results/coccinelle/bin/sp*
./results/coccinelle/bin/spatch:
	linux-vdso.so.1 (0x00007fff92710000)
	libpcre.so.1 => /nix/store/5158ikyyda1y8z707jg8spac49i8mxa3-pcre-8.45/lib/libpcre.so.1 (0x00007f017b068000)
	libm.so.6 => /nix/store/m4g6lswi75b739cpdx8wfxlfmcazyks9-glibc-2.34-115/lib/libm.so.6 (0x00007f017af8f000)
	libpthread.so.0 => /nix/store/m4g6lswi75b739cpdx8wfxlfmcazyks9-glibc-2.34-115/lib/libpthread.so.0 (0x00007f017af8a000)
	libc.so.6 => /nix/store/m4g6lswi75b739cpdx8wfxlfmcazyks9-glibc-2.34-115/lib/libc.so.6 (0x00007f017ad8b000)
	/nix/store/m4g6lswi75b739cpdx8wfxlfmcazyks9-glibc-2.34-115/lib/ld-linux-x86-64.so.2 => /nix/store/m4g6lswi75b739cpdx8wfxlfmcazyks9-glibc-2.34-115/lib64/ld-linux-x86-64.so.2 (0x00007f017b0e2000)
./results/coccinelle/bin/spgen:
	linux-vdso.so.1 (0x00007fff9fb31000)
	libpcre.so.1 => /nix/store/5158ikyyda1y8z707jg8spac49i8mxa3-pcre-8.45/lib/libpcre.so.1 (0x00007f50af183000)
	libm.so.6 => /nix/store/m4g6lswi75b739cpdx8wfxlfmcazyks9-glibc-2.34-115/lib/libm.so.6 (0x00007f50af0aa000)
	libpthread.so.0 => /nix/store/m4g6lswi75b739cpdx8wfxlfmcazyks9-glibc-2.34-115/lib/libpthread.so.0 (0x00007f50af0a5000)
	libc.so.6 => /nix/store/m4g6lswi75b739cpdx8wfxlfmcazyks9-glibc-2.34-115/lib/libc.so.6 (0x00007f50aeea6000)
	/nix/store/m4g6lswi75b739cpdx8wfxlfmcazyks9-glibc-2.34-115/lib/ld-linux-x86-64.so.2 => /nix/store/m4g6lswi75b739cpdx8wfxlfmcazyks9-glibc-2.34-115/lib64/ld-linux-x86-64.so.2 (0x00007f50af1fd000)
[nix-shell:~/.cache/nixpkgs-review/pr-171019-1]$ ldd ./results/coccinelle/lib/coccinelle/*.so
	linux-vdso.so.1 (0x00007ffe26711000)
	libc.so.6 => /nix/store/m4g6lswi75b739cpdx8wfxlfmcazyks9-glibc-2.34-115/lib/libc.so.6 (0x00007fa855aa1000)
	/nix/store/m4g6lswi75b739cpdx8wfxlfmcazyks9-glibc-2.34-115/lib64/ld-linux-x86-64.so.2 (0x00007fa855cca000)
```

Although it should not be used as a source of the truth, the deps list match up with https://aur.archlinux.org/packages/coccinelle package in aur

Reply:
I would expect pcre to not be a direct dependency but rather a transitive dependency via `ocaml_pcre`.

And as you note AUR packages are not usually a good source for dependencies for projects in languages that have their own package managers, since they are usually built unsandboxed and tend to fetch whatever.

Reply:
Looking at https://github.com/coccinelle/coccinelle/blob/master/coccinelle.opam, I would go with something like:

```patch
diff --git a/pkgs/development/tools/misc/coccinelle/default.nix b/pkgs/development/tools/misc/coccinelle/default.nix
index 4ea90873379..2b0350c4418 100644
--- a/pkgs/development/tools/misc/coccinelle/default.nix
+++ b/pkgs/development/tools/misc/coccinelle/default.nix
@@ -23,23 +23,26 @@ stdenv.mkDerivation rec {
 
   nativeBuildInputs = with ocamlPackages; [
     autoreconfHook
-    ncurses
     pkg-config
-    stdcompat
     findlib
     ocaml
-    ocaml_pcre
     menhir
-    parmap
   ];
 
   buildInputs = with ocamlPackages; [
-    pcre
-    python3
     ocaml
-    findlib
+    ocaml_pcre
+    parmap
+    pyml
+    python3
+    stdcompat
   ];
 
+  postPatch = ''
+    # Ensure dependencies from Nixpkgs are picked up.
+    rm -r bundles/
+  '';
+
   meta = {
     description = "Program to apply semantic patches to C code";
     longDescription = ''
```

Also we should drop the bundles directory to ensure packages from Nixpkgs are used: https://github.com/coccinelle/coccinelle/tree/master/bundles
And ncurses look like they were only needed by python in the past.